### DISCUSSION TITLE
Generate from nothing

### DISCUSSION THREAD

**Discussion by MikeSchumacherCapgemini** [devonfw/cobigen#1413] [Ideas]
As a Cobigen user, I want to be able to generate from no input file so that i can use Cobigen to start from scratch, while having no pre-configured files.

An example of an implementation could be to create a dialogue box asking for relevant information that would usually be taken from an input file, generating what would usually be the input file in addition to the other generation results.

This is a workshop topic, any ideas are welcome and can be discussed in the comments

**Comment by hohwille** [comment]
My thoughts on this:
* We need a declarative description of (required) variables (e.g. `name=component, type=String, description="Please enter component name` - may even include a default value or a validating regex) for such a generation increment (e.g. "Generate new component").
* CobiGen needs to traverse all of these variables and see which of them are not yet populated (from context, cobigen.properites, etc.).
* All variables not populated need to be filled by the user. In Eclipse via UI popup dialog and in CLI via commandline propts. For CLI any propery can also be provided directly as CLI arg.
* After the variables are populated, cobigen has them as its varibles and can proceed as usual with template generation.

Story sounds easy to implement but creates high value as this use-case is highly request.

**Comment by jan-vcapgemini** [comment]
Could be combined with this: https://github.com/devonfw/cobigen/issues/192
